### PR TITLE
Cirrus: Minor cleanup of dependencies and docs

### PR DESCRIPTION
**Pull request by cevich**
:smile: Actually, just docs and task-dependency cleanup then removing a long-disused "optional" task.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -508,11 +485,13 @@ success_task:
...
-        - "build_each_commit"
         - "vendor"
         - "varlink_api"
+        - "build_each_commit"
+        - "meta"
         - "testing"
-        - "special_testing"
+        - "special_testing_rootless"
+        - "special_testing_in_podman"
         - "optional_testing"
```
Comment (mheon) [inline .cirrus.yml]: Didn't you just remove this above?

Comment (cevich) [reply .cirrus.yml]: Oh, well, I guess you'll have to open up a PR to fix ***MY*** mess then :stuck_out_tongue:

Comment (cevich) [reply .cirrus.yml]: okay, fixed and pushed.  thx :smile:

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but the merge bug has bit ya @cevich

**Comment by cevich** [comment]
Seems I'm my own worst enemy lol.  No worries, I'll need to rebase this again after other PRs of mine merge as well.

**Pull request merged**